library/src/main/java/com/novoda/viewpageradapter/ViewPagerAdapter.java

```diff
@@ -89,10 +110,21 @@ public void destroyItem(ViewGroup container, int position, Object key) {
 
     private void saveViewState(int position, V view) {
         SparseArray<Parcelable> viewState = new SparseArray<>();
-        view.saveHierarchyState(viewState);
+        saveHierarchyState(view, position, viewState);
```

Comment:
we don't need position, but I suppose it's useful to anyone that would need to override this? same for restore.